### DISCUSSION TITLE
Is it possible to access the LambdaContext when using batch processing?

### DISCUSSION THREAD

**Discussion by mew1033** [aws-powertools/powertools-lambda-python#1238] [Q&A] [answered]
If so, how? I followed the docs and my code now basically looks like this:

```python
processor = BatchProcessor(event_type=EventType.SQS)


def record_handler(record: SQSRecord):
    """
    Handler for a single SQS record.
    """
    # Do some cool things with record.
    pass


@logger.inject_lambda_context
@event_source(data_class=SQSEvent)
@batch_processor(record_handler=record_handler, processor=processor)
def lambda_handler(event: Dict[str, SQSEvent], context: LambdaContext) -> Dict[str, Any]:
    return processor.response()
```

Is there any way to access the `context` object in the `record_handler` function? I'm having a hard time searching for the answer because every time I mention context and batch processing, all I find is how to do batch processing with the context manager method.

**Comment by michaelbrewer** [comment]
A work around would look something like this:

```python
processor = BatchProcessor(event_type=EventType.SQS)


def lambda_handler(event, context: LambdaContext):
    def record_handler(record: SQSRecord):
        print(context.function_name)
        ...

    with processor(records=event["Records"], handler=record_handler):
        processor.process()

    return processor.response()
```

FYI, your code example typing for `event`  should be `SQSEvent`

**Reply by mew1033** [reply]
Ah, I guess that'd work. I do wish there was a way to pass the LambdaContext into the record_handler directly.

And thanks for the tip on the event typing. I definitely missed that.

**Comment by heitorlessa** [comment]
Hey @mew1033, What information are you looking to use within Lambda Context?

Lambda runtime makes key Lambda Context data [available as environment variables](https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-runtime). If these are insufficient, please open up a [feature request](https://github.com/awslabs/aws-lambda-powertools-python/issues/new?assignees=&labels=feature-request%2Ctriage&template=feature_request.yml&title=Feature+request%3A+TITLE) as it might be an useful exercise to think of ways to expose Lambda Context more generally across any utility.

Thank you!

**Reply by mew1033** [reply]
I'm looking to use the `get_remaining_time_in_millis` function specifically. In this case, the reason is that my batch size sometimes includes very large items that might time out if they are at the end of the batch. I want to add something to the top of the record_handler that checks the remaining time, and if it's below a threshold it just raises an exception for that record. Then the partial failure mechanism can put the failed items back on the queue.

As it is right now, if the function just times out, everything is put back, not just the failed items.

**Reply by michaelbrewer** [reply]
makes sense.

**Comment by heitorlessa** [comment, marked as answer]
Closing as this was released back then: https://docs.powertools.aws.dev/lambda/python/latest/utilities/batch/#accessing-lambda-context
